Resources/config/services.xml

```diff
@@ -8,8 +8,31 @@
             <tag name="kernel.event_subscriber"/>
         </service>
 
-        <!-- enable autowiring compatibility in Symfony 3.3+ -->
-        <service id="Http\Client\HttpClient" alias="httplug.client" public="false" />
+        <!-- Discovery with autowiring support for Symfony 3.3+ -->
+        <service id="httplug.message_factory.default" class="Http\Message\MessageFactory">
+            <factory class="Http\Discovery\MessageFactoryDiscovery" method="find" />
+        </service>
+        <service id="Http\Message\MessageFactory" alias="httplug.message_factory.default" public="false" />
+
+        <service id="httplug.stream_factory.default" class="Http\Message\StreamFactory">
+            <factory class="Http\Discovery\StreamFactoryDiscovery" method="find" />
+        </service>
+        <service id="Http\Message\StreamFactory" alias="httplug.stream_factory.default" public="false" />
+
+        <service id="httplug.uri_factory.default" class="Http\Message\UriFactory">
+            <factory class="Http\Discovery\UriFactoryDiscovery" method="find" />
+        </service>
+        <service id="Http\Message\UriFactory" alias="httplug.uri_factory.default" public="false" />
```

Comment:
if we want to remove httplug.uri_factory.default in the next major version we should make the alias public instead of private so that users can be forward compatible. but i am not sure what the intention of symfony itself is on that question... so lets merge this as is and try to find out what to do for the future.